src/EventTeachingService.h

```diff
@@ -28,8 +28,10 @@ class EventTeachingService : public Service
 private:
   Controller *controller;
   Configuration *module_config;  // Shortcut to reduce indirection code.
+  byte (*checkProduced)();
```

Comment:
You need to describe what checkProduced() is used for. 
If I read the code correct and based on our chat on Tuesday it looks like this call-back function is used when FCU is teaching an event and it needs to know which input this event is for.
Maybe call it checkInputProduced() ?